File: app/src/main/java/ru/job4j/todolist/store/IStore.java
package ru.job4j.todolist.store;

import java.util.List;

import ru.job4j.todolist.Task;

public interface IStore {

    List<Task> getTasks();

    List<Task> getFilteredTasks(String filter);

    void addTask(String name,String description,String created, String photo);

    void editTask(int id, String name,String description,String closed, String photo);

    void closeOrReopenTask(int id, String closed);

    int deleteTask(int id);

    void deleteAllTasks();

    Task findTaskByID(int id);

    int getPositionOfTaskById(int id);
}

File: app/src/main/java/ru/job4j/todolist/store/IStore.kt
package ru.job4j.todolist.store

import ru.job4j.todolist.Task

interface IStore {
    fun getTasks(): List<Task>
    fun getFilteredTasks(filter: String): List<Task>
    fun addTask(name: String, description: String, created: String, photo: String?)
    fun editTask(id: Int, name: String, description: String, closed: String, photo: String)
    fun closeOrReopenTask(id: Int, closed: String?)
    fun deleteTask(id: Int): Int
    fun deleteAllTasks()
    fun findTaskByID(id: Int): Task?
}